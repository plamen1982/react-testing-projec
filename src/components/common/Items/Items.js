import React from 'react';

const Items = ({ items = [] }) => {
  if (items.length === 0) {
    return <span className="empty-message">No items in the list</span>;
  }
  if (items.length === 1) {
    return <span className="empty-message">One item: {items[0]}</span>;
  }

  return (
    <ul>
      {items.map(item => (
        <li>{item}</li>
      ))}
    </ul>
  );
};

export default Items;
