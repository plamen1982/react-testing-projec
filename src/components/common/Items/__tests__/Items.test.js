import React from 'react';
import renderer from 'react-test-renderer';

import Items from '../Items';

describe('<Items />', () => {
  test('<Items /> render ul list with three items: one, two and three,', () => {
    const items = ['one', 'two', 'three'];
    const tree = renderer.create(<Items items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<Items />, rendered only with one item', () => {
    const item = ['one'];
    const tree = renderer.create(<Items items={item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<Items /> render default message', () => {
    const tree = renderer.create(<Items />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
