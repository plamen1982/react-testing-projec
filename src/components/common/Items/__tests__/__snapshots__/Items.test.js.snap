// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Items /> <Items /> render default message 1`] = `
<span
  className="empty-message"
>
  No items in the list
</span>
`;

exports[`<Items /> <Items /> render ul list with three items: one, two and three, 1`] = `
<ul>
  <li>
    one
  </li>
  <li>
    two
  </li>
  <li>
    three
  </li>
</ul>
`;

exports[`<Items /> <Items />, rendered only with one item 1`] = `
<span
  className="empty-message"
>
  One item: 
  one
</span>
`;
